cli/tests/common/git.rs

```diff
@@ -0,0 +1,348 @@
+
+fn git_config() -> Vec<bstr::BString> {
+    vec![
+        format!("user.name = {GIT_USER}").into(),
+        format!("user.email = {GIT_EMAIL}").into(),
+        "init.defaultBranch = master".into(),
+    ]
+}
+
+fn open_options() -> gix::open::Options {
+    gix::open::Options::isolated().config_overrides(git_config())
+}
+
+pub fn open(directory: impl Into<PathBuf>) -> gix::Repository {
+    gix::open_opts(directory, open_options()).unwrap()
+}
+
+pub fn init(directory: impl AsRef<Path>) -> gix::Repository {
+    gix::ThreadSafeRepository::init_opts(
+        directory,
+        gix::create::Kind::WithWorktree,
+        gix::create::Options::default(),
+        open_options(),
+    )
+    .unwrap()
+    .to_thread_local()
+}
+
+pub fn init_bare(directory: impl AsRef<Path>) -> gix::Repository {
+    gix::ThreadSafeRepository::init_opts(
+        directory,
+        gix::create::Kind::Bare,
+        gix::create::Options::default(),
+        open_options(),
+    )
+    .unwrap()
+    .to_thread_local()
+}
+
+pub fn clone(dest_path: &Path, url: &str) -> gix::Repository {
+    let mut prepare_fetch = gix::clone::PrepareFetch::new(
+        url,
+        dest_path,
+        gix::create::Kind::WithWorktree,
+        gix::create::Options::default(),
+        open_options(),
+    )
+    .unwrap();
+    let (mut prepare_checkout, _outcome) = prepare_fetch
+        .fetch_then_checkout(gix::progress::Discard, &gix::interrupt::IS_INTERRUPTED)
+        .unwrap();
+    let (repo, _outcome) = prepare_checkout
+        .main_worktree(gix::progress::Discard, &gix::interrupt::IS_INTERRUPTED)
+        .unwrap();
+
+    repo
+}
+
+pub struct CommitResult {
+    pub tree_id: gix::ObjectId,
+    pub commit_id: gix::ObjectId,
+}
+
+pub fn add_commit(
+    repo: &gix::Repository,
+    reference: &str,
+    filename: &str,
+    content: &[u8],
+    message: &str,
+    parents: impl IntoIterator<Item = impl Into<gix::ObjectId>>,
+) -> CommitResult {
+    let blob_oid = repo.write_blob(content).unwrap();
+
+    let parents: Vec<gix::ObjectId> = parents.into_iter().map(|x| x.into()).collect();
+
+    let parent_tree_editor = parents.first().map(|commit_id| {
+        repo.find_commit(*commit_id)
+            .unwrap()
+            .tree()
+            .unwrap()
+            .edit()
+            .unwrap()
+    });
+    let head_tree_editor = get_head_tree(repo).map(|tree| tree.edit().unwrap());
+    let empty_tree_editor_fn = || {
+        repo.edit_tree(gix::ObjectId::empty_tree(repo.object_hash()))
+            .unwrap()
+    };
+
+    let mut tree_editor = parent_tree_editor
+        .or(head_tree_editor)
+        .unwrap_or_else(empty_tree_editor_fn);
+    tree_editor
+        .upsert(filename, gix::object::tree::EntryKind::Blob, blob_oid)
+        .unwrap();
+    let tree_id = tree_editor.write().unwrap().detach();
+    let commit_id = commit_tree(repo, reference, tree_id, message, parents);
+    CommitResult { tree_id, commit_id }
+}
+
+pub fn commit_tree(
```

Comment:
nit: rename to `commit` or `write_commit`? `commit_tree` sounds like a function that commits new tree object.